core/trino-main/src/main/java/io/trino/server/ui/OAuth2WebUiLogoutResource.java

```diff
@@ -14,32 +14,63 @@
 package io.trino.server.ui;
 
 import com.google.common.io.Resources;
+import com.google.inject.Inject;
 import io.trino.server.security.ResourceSecurity;
+import io.trino.server.security.oauth2.OAuth2Client;
 import jakarta.ws.rs.GET;
 import jakarta.ws.rs.Path;
 import jakarta.ws.rs.core.Context;
 import jakarta.ws.rs.core.HttpHeaders;
 import jakarta.ws.rs.core.Response;
 import jakarta.ws.rs.core.SecurityContext;
+import jakarta.ws.rs.core.UriBuilder;
 import jakarta.ws.rs.core.UriInfo;
 
 import java.io.IOException;
+import java.net.URI;
+import java.util.Optional;
 
+import static io.trino.server.security.ResourceSecurity.AccessType.PUBLIC;
 import static io.trino.server.security.ResourceSecurity.AccessType.WEB_UI;
 import static io.trino.server.ui.FormWebUiAuthenticationFilter.UI_LOGOUT;
+import static io.trino.server.ui.OAuthIdTokenCookie.ID_TOKEN_COOKIE;
 import static io.trino.server.ui.OAuthWebUiCookie.delete;
 import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Objects.requireNonNull;
 
 @Path(UI_LOGOUT)
 public class OAuth2WebUiLogoutResource
 {
+    private final OAuth2Client auth2Client;
+
+    @Inject
+    public OAuth2WebUiLogoutResource(OAuth2Client auth2Client)
+    {
+        this.auth2Client = requireNonNull(auth2Client, "auth2Client is null");
+    }
+
     @ResourceSecurity(WEB_UI)
     @GET
     public Response logout(@Context HttpHeaders httpHeaders, @Context UriInfo uriInfo, @Context SecurityContext securityContext)
             throws IOException
+    {
+        Optional<String> idToken = OAuthIdTokenCookie.read(httpHeaders.getCookies().get(ID_TOKEN_COOKIE));
+        URI callBackUri = UriBuilder.fromUri(uriInfo.getAbsolutePath())
+                .path("logout.html")
+                .build();
+
+        return Response.seeOther(auth2Client.getLogoutEndpoint(idToken, callBackUri).orElse(callBackUri))
```

Comment:
Is it possible to call sso logout ourselves behind the scenes instead of depending on user browser's doing that?
Otherwise we could end up in situation where the user doesn't follow that redirect for any reason and the value is still valid.

Reply:
The spec here mentions that 
> An RP requests that the OP log out the End-User by redirecting the End-User's User Agent to the OP's Logout Endpoint. This URL is normally obtained via the end_session_endpoint element of the OP's Discovery response or may be learned via other mechanisms. 

So I guess we should redirect on the User browser instead of handling it. Additionally each IdP can have its own way to redirecting the user right ?